server/dao/product.js

```diff
@@ -37,6 +38,47 @@ exports.findProducts = (db, searchString, category) => {
   return db.collection(productsCollectionName).find(query).toArray();
 };
 
+exports.findProductsByFarmerID = (db, farmerID, searchString, category) => {
+  //4 cases: 1. Category is defined, 2. stringSearch is defined,
+  //3.both stringSearch and Category are defined, 4. none of them are defined
+
+  let query = {};
+
+  if (searchString && category) {
+    query = {
+      $and: [
+        { farmerID: ObjectID(farmerID) },
+        { $text: { $search: searchString.toString() } },
+        { category: category.toString() },
+      ],
+    };
+  } else if (category) {
+    query = {
+      $and: [
+        { farmerID: ObjectID(farmerID) },
+        { category: category.toString() },
+      ],
+    };
+  } else if (searchString) {
+    query = {
+      $and: [
+        { farmerID: ObjectID(farmerID) },
+        { $text: { $search: searchString.toString() } },
+      ],
+    };
+  } else {
+    query = { farmerID: ObjectID(farmerID) };
+  }
+
```

Comment:
to create the query this way is a little bit confusing.
```
query="{$and:[ {farmerID:ObjectID(famrerID)},";
if(category)
query+="{ category: category.toString() },";
```
etc

Reply:
I've tried but it doesn't work